src/createAsyncThunk.ts

```diff
@@ -0,0 +1,183 @@
+import { Dispatch } from 'redux'
+import nanoid from 'nanoid'
+import { createAction } from './createAction'
+
+export type BaseThunkAPI<S, E, D extends Dispatch = Dispatch> = {
+  dispatch: D
+  getState: () => S
+  extra: E
+  requestId: string
+  signal: AbortSignal
+}
+
+/**
+ * @alpha
+ */
+export interface SerializedError {
+  name?: string
+  message?: string
+  stack?: string
+  code?: string
+}
+
+const commonProperties: (keyof SerializedError)[] = [
+  'name',
+  'message',
+  'stack',
+  'code'
+]
+
+// Reworked from https://github.com/sindresorhus/serialize-error
+export const miniSerializeError = (value: any): any => {
+  if (typeof value === 'object' && value !== null) {
+    const simpleError: SerializedError = {}
+    for (const property of commonProperties) {
+      if (typeof value[property] === 'string') {
+        simpleError[property] = value[property]
+      }
+    }
+
+    return simpleError
+  }
+
+  return value
+}
+
+/**
+ *
+ * @param type
+ * @param payloadCreator
+ *
+ * @alpha
+ */
+export function createAsyncThunk<
+  Returned,
+  ThunkArg = void,
+  State = unknown,
+  Extra = unknown,
+  DispatchType extends Dispatch = Dispatch,
```

Comment:
These expressions, and their use in the `ThunkAPI` parameter may cause a problem. Not 100% certain on this, but I think I hit this problem when I tried implementing this pattern with a more complex generic type inside fastify